porcupine/plugins/filetypes.py

```diff
@@ -252,6 +253,30 @@ def parse_filetype_name(name: str) -> FileType:
     )
 
 
+filetypes_var: tkinter.StringVar
+
+
+def _sync_filetypes_menu(event: object) -> None:
+    tab = get_tab_manager().select()
+    if isinstance(tab, tabs.FileTab):
+        try:
+            filetype_name = tab.settings.get("filetype_name", str)
+            if filetypes_var and filetype_name:
+                filetypes_var.set(filetype_name)
+        except KeyError:
+            filetypes_var.set("")
+    else:
+        filetypes_var.set("")
+
+
+def _add_filetype_menuitem(name: str, tk_var: tkinter.StringVar) -> None:
+    menubar.get_menu("Filetypes").add_radiobutton(
+        label=name,
+        command=lambda: apply_filetype_to_tab(filetypes[name], menubar.get_filetab()),
```

Comment:
`menubar.get_filetab()` fails if the selected tab is not a `FileTab`. Previously this wasn't a problem, because `add_filetab_command()` includes a call to `set_enabled_based_on_tab()`.

The simplest solution would be disabling the entire "Filetypes" menu, unless you are currently in a `FileTab`, but that's not ideal because then you can't click the "Edit filetypes.toml" button at the bottom. A slightly better solution would be to call `set_enabled_based_on_tab()` in this function, so that every filetype menuitem gets disabled just like before.

Reply:
I noticed this as well - when I click on Filetypes -> Edit filetypes.toml:
I do get a callback to _sync_filetypes_menu and can lookup the Tab but the settings does not contain a filetype_name (or much else)

After I select another tab and then select the tab again it does contain the filetype_name. 
So in between my first callback and my second callback for filetypes.toml the settings data gets populated.
I'm just not sure where that is happening. 
Ideally, I'd get my first callback after whatever is populating the tab settings has finished it's work. Is that possible?